app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java

```diff
@@ -57,36 +61,90 @@ public void init() {
     @Override
     public void setListeners() {
         signupButton.setOnClickListener(v -> {
-            Intent intent = new Intent(getActivity(), SignupActivity.class);
-            startActivity(intent);
-
+            this.clearSetErrorMessages();
+            if (this.mailIsCorrect()) {
+                Intent intent = new Intent(getActivity(), SignupActivity.class);
+                startActivity(intent);
+            }
         });
-
         loginButton.setOnClickListener(v -> {
+            this.clearSetErrorMessages();
             if (this.mailAndPasswordInputAreCorrect()) {
-                getPresenter().storeUser(mailInput.getText().toString(), passwordInput.getText().toString(), getActivity());
-                Intent intent = new Intent(getActivity(), HomeActivity.class);
-                startActivity(intent);
+                getPresenter().getUserByMail(mailInput.getText().toString(), passwordInput.getText().toString(), getActivity());
             }
         });
+        conditionsTextView.setOnClickListener(v -> {
```

Comment:
I think we change this in the frontend. You should rebase this branch with development